### PR TITLE
[CI Visibility] - Add a new helper method to the discovery service

### PR DESCRIPTION
**Pull request by tonyredondo**
## Summary of changes

This PR refactors the use of the DiscoveryService in CI Visibility.

## Reason for change

Suggestions of #3433

### CODE REVIEW

File: tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs [hunk 1]
```diff
@@ -82,6 +82,24 @@ public static DiscoveryService Create(ImmutableExporterSettings exporterSettings
                     () => new MinimalAgentHeaderHelper(),
                     uri => uri));
 
+        public static DiscoveryService Create(
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs:85]: Can we combine this with the Create function above? Maybe move the default values for the retries etc to this method, and get rid of the second constructor?

**Review by andrewlock — APPROVED** [review]
Thanks!

**Review by pierotibou — APPROVED** [review]

File: tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs:83]: I removed the .ctor, but I did not combine both `Create` functions, there's a TimeSpan parameter that we cannot have as a default value (but maybe as an optional parameter???)

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:warning: Merging #3441 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e4300cf64aefe69c510ef0c625dd2341ed266a85) will will **decrease** line coverage by `1%`
:warning: Merging #3441 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e4300cf64aefe69c510ef0c625dd2341ed266a85) will will **decrease** branch coverage by `1%`
:heavy_check_mark: Merging #3441 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e4300cf64aefe69c510ef0c625dd2341ed266a85) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/e4300cf64aefe69c510ef0c625dd2341ed266a85) | #3441       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `20376` / `27674` | `20223` / `27681` |          |
| Lines %   | `74%`      | `73%`      |  `-1%` :warning:  |
| Branches  | `12039` / `17278` | `11947` / `17278` |          |
| Branches %| `70%`      | `69%`      |  `-1%` :warning:  |
| Complexity|   `18687`      | `18687`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=117977&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/117977/artifacts?artifactName=Code%20Coverage%20Report_117977&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3441](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=118020&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/118020/artifacts?artifactName=Code%20Coverage%20Report_118020&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :warning:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/e4300cf64aefe69c510ef0c625dd2341ed266a85) | #3441       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `74%`      | `73%`       |  `-1%` :warning:  |
| Branches %| `70%`      | `69%`       |  `-1%` :warning:  |
| Complexity| `18687`      | `18687`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Agent.TraceSamplers.AnalyticsEventsSampler](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Agent/TraceSamplers/AnalyticsEventsSampler.cs) | `-100%` :no_entry: | `-100%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.TraceSamplers.ErrorSampler](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Agent/TraceSamplers/ErrorSampler.cs) | `-100%` :no_entry: | `-100%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.TraceSamplers.PrioritySampler](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Agent/TraceSamplers/PrioritySampler.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Sampling.GlobalSamplingRule](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Sampling/GlobalSamplingRule.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.Transports.JsonTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Telemetry/Transports/JsonTelemetryTransport.cs) | `-33%` :no_entry: | `-42%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.DirectSubmission.DirectLogSubmissionManager](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Logging/DirectSubmission/DirectLogSubmissionManager.cs) | `-33%` :no_entry: | `-33%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.DirectSubmission.Sink.NullDatadogSink](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Logging/DirectSubmission/Sink/NullDatadogSink.cs) | `-25%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Processors.NormalizerTraceProcessor](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Processors/NormalizerTraceProcessor.cs) | `-22%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `-17%` :no_entry: | `-11%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Processors.ObfuscatorTraceProcessor](https://github.com/DataDog/dd-trace-dotnet/tree/ef8c6449ebfcafa49af997d093e4f242a395cf77/tracer/src/Datadog.Trace/Processors/ObfuscatorTraceProcessor.cs) | `-12%` :no_entry: | `-15%` :no_entry:   | `0` :heavy_check_mark: |
| ...And 4 more  | | | |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=117977&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/117977/artifacts?artifactName=Code%20Coverage%20Report_117977&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3441](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=118020&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/118020/artifacts?artifactName=Code%20Coverage%20Report_118020&api-version=6.0&%24format=zip)

**Pull request merged**